src/concrete/ml/pytest/torch_models.py

```diff
@@ -1409,3 +1409,29 @@ def forward(self, x):
         """
 
         return self.sub_module(x)
+
+
+class EncryptedMatrixMultiplicationModel(nn.Module):
+    """PyTorch module for performing matrix multiplication between to encrypted values."""
+
+    # pylint: disable-next=unused-argument
+    def __init__(self, input_output, activation_function):
+        super().__init__()
+        self.act = activation_function()
+
+    def forward(self, input1):
+        """Forward function for matrix multiplication.
+
+        Args:
+            input1 (torch.Tensor): The first input tensor.
+
+        Returns:
+            torch.Tensor: The result of the matrix multiplication.
+        """
+        input1 = input1.unsqueeze(1)
+
+        output = torch.matmul(input1, input1.transpose(2, 1))
+        output = self.act(output)
+
+        output = output.squeeze(-1)  # Squeezes out one of the last two singleton dimensions
```

Comment:
small detail but I think we prefer to put comments above the line instead of having them inline like this

Reply:
Not sure if we have a rule on this 🤷🏼

Reply:
Alright I will put it above the line. Should be better to match our 100 chars limit.